### DISCUSSION TITLE
PDM stopped working with Pyenv out of the blue

### DISCUSSION THREAD

**Discussion by robml** [pdm-project/pdm#951] [Q&A]
After a lot of troubleshooting to properly work with PDM I thought I had a good workflow going. However recently (with no changes except an update to PDM itself) the entire package stopped working. How? Well I suspect it has to do with PyEnv. For context I am developing using WSL2 (Ubuntu on Windows) and it has served me well so far. As you know Ubuntu comes with a default Python of 3.8, however for development and production purposes I use 3.9.10 and 3.10.2, for which Pyenv seemed like a good solution.

So I initialized my projects with either version and that was that. Now however coming back, I can no longer seem to work with any of my former projects, even typing something as simple as `pdm info` brings the following message

`[InvalidPyVersion]: Invalid Python interpreter: /home/robmel/.pyenv/shims/python`

I want to know if anyone else has faced this and how they've dealt with it, I don't want to report as a bug if there isn't one.

**Comment by pawamoy** [comment]
Did you try to uninstall and reinstall PDM and/or pyenv versions?

**Reply by robml** [reply]
I went back to try and do just that, and discover now that my entire kernel is messed up. I might have to re-install WSL2 all over again.

**Reply by AdamJel** [reply]
Did it work? Might you "close" this question?
